provider/tags.go

```diff
@@ -57,6 +57,40 @@ func applyTags(
 	}
 	if allTags.IsNull() {
 		delete(ret, "tags")
+		delete(ret, "tagsAll")
+		return ret, nil
+	}
+	ret["tags"] = allTags
```

Comment:
Why do we set `allTags` to `tags`? Shouldn't `tags` only include the tags specified on the resource?

Reply:
Yeah I was also confused here, I was expecting something like this:

- (1) we receive data from the Read call that has "tags" and "tags_all"

- (2) I think here we need to understand what this data is and do we trust it; e.g. the TF provider would be grabbing all tags from the AWS hopefully, but then it would also possibly applying the defaultTags to that result to either union or exclude them? We had to disable this behavior for Create/Update lifecycle since we couldn't get the implementation to behave correctly and trust it; perhaps something similar needs to be tweaked here for reads 

- (3) suppose we figured out (2) correctly and we obtained what we believe is the set of in-AWS resource tags. Now what do we decide to surface as "tags". Perhaps here we want to exclude surfacing a k:v pair in "tags" if the defaultTags already has it set at provider level. So perhaps take the set difference here?

Unfortunately it's more complicated than this as TransformOutputs hook is hiding some differences around the context this code is executing in. Let me add below.